compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala

```diff
@@ -49,7 +49,7 @@ trait QuotesAndSplices {
       report.error(em"Quotes require stable Quotes, but found non stable $qctx", qctx.srcPos)
 
     if ctx.mode.is(Mode.Pattern) then
-      typedQuotePattern(tree, pt, qctx).withSpan(tree.span)
+      makeInlineable(typedQuotePattern(tree, pt, qctx).withSpan(tree.span))
```

Comment:
```suggestion
      PrepareInlineable.makeInlineable(typedQuotePattern(tree, pt, qctx).withSpan(tree.span))
```
No real idea why but ignoring the Context switch from makeInlinleable (and instead using PrepareInlineable.makeInlineable directly) seems to fix both the dottytags and quill regressions minimalisations